### ISSUE TITLE
High DPI Support

### ISSUE THREAD

**Issue by thejustinwalsh** [HeapsIO/heaps#589]
Requesting High DPI support.

Looks like all the work in the hashlink SDL plugin is already done to support passing the `SDL_WINDOW_ALLOW_HIGHDPI` flag at window creation time. Not sure what the required support would look like when using DirectX.

Would love to see the feature abstracted through heaps and any advice on best implementation is welcome as I can submit a PR to support it.

**Comment by thejustinwalsh** [comment]
Notes:
On iOS you must do two things to get a high DPI rendering context.
1.  NSHighResolutionCapable = YES in Info.plist
2. Pass in the `SDL_WINDOW_ALLOW_HIGHDPI` flag when creating the window by changing https://github.com/HeapsIO/heaps/blob/170bb54bdff9008420e23de5db132aaa1b5c6a62/hxd/Window.hl.hx#L42  
to  (formatted to not wrap)
```haxe
window = new sdl.Window(title, width, height, 
  sdl.Window.SDL_WINDOWPOS_CENTERED, sdl.Window.SDL_WINDOWPOS_CENTERED,
  sdl.Window.SDL_WINDOW_SHOWN | 
  sdl.Window.SDL_WINDOW_RESIZABLE | 
  sdl.Window.SDL_WINDOW_ALLOW_HIGHDPI
);
```

From what I can tell if you Omit the NSHighResolutionCapable flag or set it to NO in your Info.plist the additional `SDL_WINDOW_ALLOW_HIGHDPI` seems to have no effect. Although I am not certain at this time of how this flag affects all platforms. Also worth noting, the calls to `hxd.System.width` and `hxd.System.height` will return the size of the device screen in points and you would need to know the DPI of your screen to apply the appropriate scale from points to pixels when running in High DPI mode.

**Comment by ncannasse** [comment]
In order to support this on SDL first, we would need to be able to query from SDL for the screen DPI, as described here https://nlguillemot.wordpress.com/2016/12/11/high-dpi-rendering/

This requires adding some new primitives to Hashlink/SDL support here:
https://github.com/HaxeFoundation/hashlink/tree/master/libs/sdl
and Haxe wrapper
https://github.com/HaxeFoundation/hashlink/tree/master/libs/sdl/sdl

Once this is done, I guess we could enable it with a define:  -D highdpi and add the #if highdpi sections in the SDL Window support.

**Comment by AustinEast** [comment]
Looks like this issue is fixed if you use the latest version of SDL (beyond the latest release on homebrew). This is a quick fix that has worked for me and others:

1. Grab a newer version of SDL and replace your current installation with it (this mirror works fine: https://github.com/spurious/SDL-mirror)
2. Rebuild hashlink (with the normal `make` -> `make install` instructions)
3. Rebuild your project and see the fixed display!
